.github/workflows/push.yml

```diff
@@ -25,15 +47,47 @@ jobs:
       - name: Build project
         run: ~/.elan/bin/lake -Kenv=dev build LeanAPAP
 
+      - uses: actions/cache@v3
+        name: Mathlib doc Cache
+        with:
+          path: build/doc/Mathlib
+          key: DocGen4-${{ hashFiles('lake-manifest.json') }}
+          restore-keys: |
+            DocGen4-
+
       - name: Build documentation
         run: ~/.elan/bin/lake -Kenv=dev build LeanAPAP:docs
 
-      - name: Copy documentation
-        run: cp -R build/doc docs/
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip' # caching pip dependencies
+
+      - name: Install blueprint apt deps
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y graphviz libgraphviz-dev pdf2svg dvisvgm texlive-full
+
+      - name: Install blueprint deps
+        run: |
+          cd blueprint && pip install -r requirements.txt
+
+      - name: Build blueprint and copy to `docs/blueprint`
+        run: |
+          inv all
+
+      - name: Copy documentation to `docs/docs`
+        run: |
+          mv build/doc docs/docs
 
       - name: Remove .gitignore for gh-pages
         run: rm docs/.gitignore
 
+      - name: Count sorries
+        run: |
+          python scripts/count_sorry.py
```

Comment:
Previously this was in `inv ci` but most of the tasks in it are for Lean 3 and obsolete, I've moved it here. Feel free to add steps you need back in `inv ci`.